src/api.ts

```diff
@@ -571,7 +571,7 @@ export async function proxyRequestToContainer( req: any, res: any, container: Co
  * Pulls an image. Calls onProgress() when there is an update, resolves the returned promise
  * when the image is pulled
  */
-export async function pullImage( imageName: ImageName, onProgress: ( data: any ) => void ) {
+export async function pullImage( imageName: ImageName, onProgress: ( data: any ) => void ):Promise<void> {
```

Comment:
Not related to this change, but my VSCode was refusing to start the project without this fix.

Reply:
Interesting. This function returns the result of `findContainer` which should be `ContainerInfo` not `void`.

Reply:
I don't see that. It returns a Promise (line 582) that is resolved with no value (line 590)
Am I reading it wrong?

Reply:
Oh nevermind I was looking at the `createContainer` function 🙁 Diffs can be confusing.

There's still something suspicious in `pullImage` though:
```js
new Promise( async ( resolve, reject ) => {
  const resolvedStream = await stream;
  /* ... */
} );
```

If `await stream` fails, the error will be forgotten and unhandled. Because the promise executor returns a rejected promise. And the return value of a promise executor function is ignored. It would have to throw to make the promise reject.

By the way, I just tested that TypeScript is unable to infer the type of the `Promise` from the type of `resolve` argument. One would expect that `p` is inferred to be `Promise<number>`:
```js
const p = new Promise( resolve => { resolve( 1 ); } );
```
But it's `Promise<unknown>`.

That's why `pullImage` return value needs to be typed explicitly. Alternatively, `new Promise<void>( ... )` could also work.